### ISSUE TITLE
Take screenshots of running game

### ISSUE THREAD

**Issue by leafo** [itchio/itch#535]
This is part of a much bigger project, but I think it would be awesome to have an integrated screenshot system that lets people upload screenshots to their itch.io pages.

**Comment by fasterthanlime** [comment]
I've looked into it, and I'm not happy with 'only screenshots'. Screenshots are easy to do, but I really want to see a game in motion, so it's video or nothing.

Electron has a desktopCapturer: https://github.com/electron/electron/blob/master/docs/api/desktop-capturer.md - but I seriously doubt performance will be good enough, and there doesn't seem to be a way to determine which window is the right one to capture.

I've started looking into [OBS studio](https://github.com/jp9000/OBS-studio) (libobs in particular), unsuccesfully tried to compile it on Windows (MSVC 2015), my impressions are: it's a rewrite, which is great, it's flexible, which is great, it's up-to-date, that's cool, but it has MANY dependencies, which isn't that great, at all. It's a large project that, for example, insists on interfacing directly with ffmpeg as a library (rather than piping data to it for example).

I've also looked into a much older project, taksi ([own fork here](https://github.com/fasterthanlime/taksi) - original is on SourceForge :older_man:), which stopped at DirectX9, has a crappy Windows UI, supports Windows 95 and made the only OpenGL game I could find bug like there was no tomorrow - but I think their approach is sound, and I'd rather have us have our own, small codebase that works on OpenGL/DirectX9-10-11 games with a minimalist overlay that shows recording status.

_TL;DR if we want to make the vine of video games we'll have to code it ourselves - but the basic ideas are here_
